File: iteration2/CSE3063F21P1_GRP16/src/main/java/Models/Course.java
package Models;
import Types.CourseType;

import java.util.*;

public class Course {

    private String courseName;
    private String courseCode;
    private int quota;
    private String prerequisiteCourse;
    private int credits;
    private Semester courseSemesterName;
    private CourseType courseType ; //1..8(Semester number),FTE,NTE,UE,TE
    private ArrayList<CourseSession> sessions= new ArrayList<CourseSession>();//includes all sessions,and their informations like day and hour
    private ArrayList<CourseEvents> courseEvents;
    private int quotaCounter;////////////////
    private int requiredCredits;/////////////////////////




    public Course(String courseName, String courseCode, int quota, String prerequisiteCourse, int credits,
                  Semester semester, CourseType courseType, ArrayList<CourseSession> sessions,int requiredCredits) {

        this.courseName = courseName;
        this.courseCode = courseCode;
        this.quota = quota;
        this.prerequisiteCourse = prerequisiteCourse;
        this.credits = credits;
        this.courseSemesterName = semester;
        this.courseType = courseType;
        this.sessions = sessions;
        this.quotaCounter=0;////////////////////
        this.requiredCredits=requiredCredits;
        this.courseEvents=new ArrayList<>();

        setEvents();
    }

    private void setEvents() {
        courseEvents.add(new E0Success());
        courseEvents.add(new E1ReTakeCourse());
        courseEvents.add(new E2prerequisite());
        courseEvents.add(new E3HourCollusion());
        courseEvents.add(new E4NoEnoughCredits());
        courseEvents.add(new E5Quota());


    }

    public ArrayList<CourseEvents> getCourseEvents() {
        return courseEvents;
    }

    //getters and setters
    public String getCourseName() {
        return courseName;
    }
    public void setCourseName(String courseName) {
        this.courseName = courseName;
    }
    public String getCourseCode() {
        return courseCode;
    }
    public void setCourseCode(String courseCode) {
        this.courseCode = courseCode;
    }
    public int getQuota() {
        return quota;
    }
    public void setQuota(int quota) {
        this.quota = quota;
    }

}

File: iteration3/CSE3063F21P1_GRP16/models/Course.py
class Course:
    def __init__(self, courseName, courseCode, quota, prerequisiteCourse, credits, semester, courseType, sessions, requiredCredits):
        self._courseName = courseName
        self._courseCode = courseCode
        self._quota = quota
        self._prerequisiteCourse = prerequisiteCourse
        self._credits = credits
        self._courseSemesterName = semester
        self._courseType = courseType
        self._sessions = sessions
        self._quotaCounter = 0
        self._requiredCredits = requiredCredits
        self._CourseEvents = []
        self.set_events()

    def set_events(self):
        self._CourseEvents.append(E0Success())
        self._CourseEvents.append(E1ReTakeCourse())
        self._CourseEvents.append(E2prerequisite())
        self._CourseEvents.append(E3HourCollusion())
        self._CourseEvents.append(E4NoEnoughCredits())
        self._CourseEvents.append(E5Quota())

    def getCourseEvents(self):
        return self._CourseEvents

    # getters and setters
    def getcourseName(self):
        return self._courseName

    def setcourseName(self, courseName):
        self._courseName = courseName

    def getCourseCode(self):
        return self._courseCode

    def setCourseCode(self, courseCode):
        self._courseCode = courseCode

    def getQuota(self):
        return self._quota

    def setQuota(self, quota):
        self._quota = quota